pkg/library/libraryapplyconfiguration/equivalence.go

```diff
@@ -84,52 +87,117 @@ func EquivalentClusterApplyResultIgnoringEvents(field string, lhs, rhs SingleClu
-		currSourceRequests := manifestclient.RequestsForResource(allSourceRequests, currMissingInDestination.GetLookupMetadata())
-		currDestinationRequests := manifestclient.RequestsForResource(allDestinationRequests, currMissingInDestination.GetLookupMetadata())
+	for _, currSourceRequest := range sourceRequestsToCheck {
+		currDestinationRequests := manifestclient.RequestsForResource(allDestinationRequests, currSourceRequest.GetLookupMetadata())
 
 		if len(currDestinationRequests) == 0 {
-			reasons = append(reasons, fmt.Sprintf("%s is missing: %v", nameOfDestination, currMissingInDestination.StringID()))
+			reasons = append(reasons, fmt.Sprintf("%s is missing: %v", nameOfDestination, currSourceRequest.StringID()))
 			continue
 		}
 
-		for _, currLHS := range currSourceRequests {
-			found := false
-			mismatchReasons := []string{}
-			for i, currRHS := range currDestinationRequests {
-				if manifestclient.EquivalentSerializedRequests(currLHS, currRHS) {
-					found = true
-					mismatchReasons = nil
-					break
-				}
-				// we know the metadata is the same, something else doesn't match
-				if !bytes.Equal(currLHS.GetSerializedRequest().Body, currRHS.GetSerializedRequest().Body) {
-					mismatchReasons = append(mismatchReasons,
-						fmt.Sprintf("mutation: %v, rhs[%d]: body diff: %v",
-							currMissingInDestination.StringID(),
-							i,
-							cmp.Diff(currLHS.GetSerializedRequest().Body, currRHS.GetSerializedRequest().Body),
-						),
-					)
-				}
-				if !bytes.Equal(currLHS.GetSerializedRequest().Options, currRHS.GetSerializedRequest().Options) {
-					mismatchReasons = append(mismatchReasons,
-						fmt.Sprintf("mutation: %v, rhs[%d]: options diff: %v",
-							currMissingInDestination.StringID(),
-							i,
-							cmp.Diff(currLHS.GetSerializedRequest().Options, currRHS.GetSerializedRequest().Options),
-						),
-					)
+		isApply := currSourceRequest.GetSerializedRequest().Action == manifestclient.ActionApply || currSourceRequest.GetSerializedRequest().Action == manifestclient.ActionApplyStatus
+		lhsMetadata := expandedMetadataFor(currSourceRequest.GetSerializedRequest())
+
+		found := false
+		mismatchReasons := []string{}
+		for i, currDestinationRequest := range currDestinationRequests {
+			if manifestclient.EquivalentSerializedRequests(currSourceRequest, currDestinationRequest) {
+				found = true
+				mismatchReasons = nil
+				break
+			}
+			// if we're doing an apply and the field manager doesn't match, then it's just a case of "content isn't here" versus a diff
+			// actions match because the metadata (which contains action) matched
+			if isApply {
+				lhsOptions := currSourceRequest.GetSerializedRequest().Options
+				rhsOptions := currDestinationRequest.GetSerializedRequest().Options
+				if !bytes.Equal(lhsOptions, rhsOptions) {
+					// if the options for apply (which contains the field manager) aren't the same, then the requests
+					// are logically different requests and incomparable
+					continue
```

Comment:
okay, so, for apply requests to be comparable, they need to come from the same field manager

Reply:
> okay, so, for apply requests to be comparable, they need to come from the same field manager

Correct.  And if there is no comparable value, then we assume it's missing so the diff "fails", but with a better message.